src/redis-benchmark.c

```diff
@@ -1594,6 +1677,7 @@ int parseOptions(int argc, char **argv) {
 "                    on the command line.\n"
 " -I                 Idle mode. Just open N idle connections and wait.\n"
 " -x                 Read last argument from STDIN.\n"
+" --readonly         Prepend the readonly command before sending any commands\n"
```

Comment:
I think if the readonly argument is what we need?
basically I would expect the test to define argument like "--test-replicas" which in case the benchmark is issuing a read command it will also issue that command to the replicas (which is the logic you implemented)

Reply:
Personally, I think the flag should remain `--readonly` to be explicit about what redis-benchmark is actually doing since technically the flag can be passed and replicas still won't be tested if you benchmark a write command. How would you feel about expanding the description of the argument to include the fact that for read commands it will result in benchmarking replicas and primaries?